### DISCUSSION TITLE
An invalid argument was supplied: listen_no_tls

### DISCUSSION THREAD

**Discussion by sx8469** [h2oai/wave#2079] [Q&A] [answered]
I just installed h20_wave and when I tried running the app or server, i was hit with this error.

2023/07/18 15:39:27 # 
2023/07/18 15:39:27 # ┌────────────────┐ H2O Wave 
2023/07/18 15:39:27 # │  ┐┌┐┐┌─┐┌ ┌┌─┐ │ 0.26.1 20230703100254
2023/07/18 15:39:27 # │  └┘└┘└─└└─┘└── │ © 2021 H2O.ai, Inc.
2023/07/18 15:39:27 # └────────────────┘
2023/07/18 15:39:27 # ┌──────────────────────────────────────┐
2023/07/18 15:39:27 # │  Running at http://localhost:10101/  │
2023/07/18 15:39:27 # └──────────────────────────────────────┘
2023/07/18 15:39:27 # {"address":":10101","base-url":"/","t":"listen","web-dir":"venv\\www"}
2023/07/18 15:39:27 # {"error":"listen tcp :10101: socket: An invalid argument was supplied.","t":"listen_no_tls"}

Can someone please make me understand what is the issue and what can I do, please?

I'm just running the command `waved.exe`.

My Dependency version:
h2o-wave==0.26.1
Python 3.9.13

**Comment by mturoci** [comment]
Hi @sx8469.

Can you outline the exact steps you took? Both for installation and running? If all you want is to run a Wave app, you need to:

```sh
pip install h2o-wave
wave run app.py
```

**Reply by sx8469** [reply]
Yes, I have did the same thing and the error i was getting was:
(venv) PS H:\Documents\repositories\llm-for-sf\venv> wave run ..\src\wave_app.py
2023/07/18 18:16:17 # 
2023/07/18 18:16:17 # ┌────────────────┐ H2O Wave 
2023/07/18 18:16:17 # │  ┐┌┐┐┌─┐┌ ┌┌─┐ │ 0.26.0 20230626122551
2023/07/18 18:16:17 # │  └┘└┘└─└└─┘└── │ © 2021 H2O.ai, Inc.
2023/07/18 18:16:17 # └────────────────┘
2023/07/18 18:16:17 # ┌──────────────────────────────────────┐
2023/07/18 18:16:17 # │  Running at http://localhost:10101/  │
2023/07/18 18:16:17 # └──────────────────────────────────────┘
2023/07/18 18:16:17 # {"address":":10101","base-url":"/","t":"listen","web-dir":"H:\\Documents\\repositories\\llm-for-sf\\venv\\www"}
2023/07/18 18:16:17 # {"error":"listen tcp :10101: socket: An invalid argument was supplied.","t":"listen_no_tls"}
Cannot connect to Wave server, retrying...
Cannot connect to Wave server, retrying...
Cannot connect to Wave server, retrying...
Could not connect to Wave server. Please start the Wave server (waved or waved.exe) prior to running any app.

**Reply by mturoci** [reply]
Why are you running it from within a `venv`?

**Reply by sx8469** [reply]
I was following this documentation: https://wave.h2o.ai/docs/installation

**Reply by mturoci** [reply]
Where does it say to cd into `venv`? Please outline all the steps you made, otherwise I will not be able to help you.

**Reply by sx8469** [reply]
It doesnt but it was trying to run the waved.exe directly as advised here in the last line of the error where i was running the wave app file:
_Could not connect to Wave server. Please start the Wave server (waved or waved.exe) prior to running any app._

**Reply by sx8469** [reply]
Sorry If I'm replying with vague or incomplete answers, I'm a bit new to this.

**Reply by mturoci** [reply]
> Sorry If I'm replying with vague or incomplete answers, I'm a bit new to this.

No worries. Create a new directory and try to do everything like before until you get stuck. Then copy the terminal output and paste it here so that I can see what you did.

> It doesnt but it was trying to run the waved.exe directly as advised here in the last line of the error where i was running the wave app file:
Could not connect to Wave server. Please start the Wave server (waved or waved.exe) prior to running any app.

I see. In that case, you need - https://wave.h2o.ai/docs/tutorial-hello#step-1-start-the-wave-server.

**Reply by sx8469** [reply]
I'm getting this error when I try running waved directly:
![image](https://github.com/h2oai/wave/assets/139866735/45c521f5-52ab-4640-b59a-95ea7dde6a2e)

**Reply by mturoci** [reply]
Read my previous comment.

**Reply by sx8469** [reply]
I'm still getting the same error after what you asked me to try:
![image](https://github.com/h2oai/wave/assets/139866735/db49a10c-ece7-48a3-838c-40fae1ff2b54)

**Reply by mturoci** [reply, marked as answer]
> I'm still getting the same error after what you asked me to try:

You are doing the same thing over and over. I asked you for complete terminal output. Instead, you sent me the same final step that you posted twice already. That's not helpful. I wanted to see how you create a virtual env, install the package etc.

I tried to simplify the docs as much as I could, hope this helps.

![image](https://github.com/h2oai/wave/assets/64769322/37ba7cc7-cdbe-4d6b-a061-9c0f698fdd92)
